airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/generate_erd/pipeline.py

```diff
@@ -0,0 +1,176 @@
+
+    async def _get_discovered_catalog(self, connector_to_discover: Container) -> str:
+        source_config_path_in_container = "/data/config.json"
+        discover_output = (
+            await connector_to_discover.with_new_file(source_config_path_in_container, contents=self._get_default_config().value)
+            .with_exec(["discover", "--config", source_config_path_in_container])
+            .stdout()
+        )
+        return self._get_schema_from_discover_output(discover_output)
+
+    def _get_default_config(self) -> Secret:
+        if not self.context.local_secret_store:
+            raise ValueError("Expecting local secret store to be set up but couldn't find it")
+
+        filtered_secrets = [secret for secret in self.context.local_secret_store.get_all_secrets() if secret.file_name == "config.json"]
+        if not filtered_secrets:
+            raise ValueError("Expecting at least one secret to match name `config.json`")
+        elif len(filtered_secrets) > 1:
+            print(f"Expecting only one secret with name `config.json but got {len(filtered_secrets)}. Will take one randomly...")
+
+        return filtered_secrets[0]
+
+    @staticmethod
+    def _get_schema_from_discover_output(discover_output: str) -> str:
+        for line in discover_output.split("\n"):
+            json_line = json.loads(line)
+            if json_line.get("type") == "CATALOG":
+                return json.dumps(json.loads(line).get("catalog"))
+        raise ValueError("No catalog was found in output")
+
+    def _build_erd_container(self, connector_directory: Directory, discovered_catalog: str) -> Container:
+        """Create a container to run ERD generation."""
+        container = with_poetry(self.context)
+        if self.context.genai_api_key:
+            container = container.with_env_variable("GENAI_API_KEY", self.context.genai_api_key.value)
+
+        container = (
+            container.with_mounted_directory("/source", connector_directory)
+            .with_new_file("/source/erd/discovered_catalog.json", contents=discovered_catalog)
+            .with_mounted_directory("/app", self.context.erd_dir)
+            .with_workdir("/app")
+        )
+
+        return container.with_exec(["poetry", "lock", "--no-update"]).with_exec(["poetry", "install"])
+
+
+class UploadDbmlSchema(Step):
+    context: ConnectorContext
+
+    title = "Upload DBML file to dbdocs.io"
+
+    def __init__(self, context: PipelineContext) -> None:
+        super().__init__(context)
+
+    async def _run(self, connector_to_discover: Container) -> StepResult:
+        if not self.context.dbdocs_token:
+            raise ValueError(
+                "In order to publish to dbdocs, DBDOCS_TOKEN needs to be provided. Either pass the value or skip the publish step"
+            )
+
+        connector = self.context.connector
+        source_dbml_content = open(_get_dbml_file(connector.code_directory)).read()
+
+        dbdocs_container = await (
```

Comment:
I'm not sure this `await` is required here. I don't see a problem in making the container evaluated on your final `.stdout()` call on line `138`.